packages/superset-ui-chart-controls/src/utils/advancedAnalytics/utils/getMetricOffsetsMap.ts

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable camelcase */
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitationsxw
+ * under the License.
+ */
+import { getMetricLabel, ensureIsArray, PostProcessingFactory } from '@superset-ui/core';
+
+export const TIME_COMPARISION = '__';
+
+export const getMetricOffsetsMap: PostProcessingFactory<Map<string, string>> = (
+  formData,
+  queryObject,
+) => {
+  const queryMetrics = ensureIsArray(queryObject.metrics);
```

Comment:
The [QueryFields](https://github.com/apache-superset/superset-ui/blob/8eac28a2190cd3b546807e1144ac3d6fc8809d44/packages/superset-ui-core/src/query/types/QueryFormData.ts#L62) in `queryObject` should already be guaranteed to be either `undefined` or an array. It was processed in `extraQueryFields.ts`.